### PR TITLE
r/devicefarm_instance_profile - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTS=TestAccDeviceFarmInstanceProfile_ PKG=devicefarm
--- PASS: TestAccDeviceFarmInstanceProfile_disappears (56.17s)
--- PASS: TestAccDeviceFarmInstanceProfile_basic (114.85s)
--- PASS: TestAccDeviceFarmInstanceProfile_tags (149.49s)
```

### CODE REVIEW

File: internal/service/devicefarm/instance_profile.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"package_cleanup": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"reboot_after_user": {
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile.go:53]: Should be `reboot_after_use`.

Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile.go:53]:
```suggestion
			"reboot_after_use": {
```

File: internal/service/devicefarm/instance_profile.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+	if v, ok := d.GetOk("exclude_app_packages_from_cleanup"); ok && v.(*schema.Set).Len() > 0 {
+		input.ExcludeAppPackagesFromCleanup = flex.ExpandStringSet(v.(*schema.Set))
+	}
+
+	if v, ok := d.GetOk("package_cleanup"); ok {
+		input.PackageCleanup = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("reboot_after_user"); ok {
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile.go:86]:
```suggestion
	if v, ok := d.GetOk("reboot_after_use"); ok {
```

File: internal/service/devicefarm/instance_profile.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+		return fmt.Errorf("error reading DeviceFarm Instance Profile (%s): %w", d.Id(), err)
+	}
+
+	arn := aws.StringValue(instaceProf.Arn)
+	d.Set("arn", arn)
+	d.Set("name", instaceProf.Name)
+	d.Set("description", instaceProf.Description)
+	d.Set("exclude_app_packages_from_cleanup", flex.FlattenStringSet(instaceProf.ExcludeAppPackagesFromCleanup))
+	d.Set("package_cleanup", instaceProf.PackageCleanup)
+	d.Set("reboot_after_user", instaceProf.RebootAfterUse)
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile.go:131]:
```suggestion
	d.Set("reboot_after_use", instaceProf.RebootAfterUse)
```

File: internal/service/devicefarm/instance_profile.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+		if d.HasChange("exclude_app_packages_from_cleanup") {
+			input.ExcludeAppPackagesFromCleanup = flex.ExpandStringSet(d.Get("exclude_app_packages_from_cleanup").(*schema.Set))
+		}
+
+		if d.HasChange("package_cleanup") {
+			input.PackageCleanup = aws.Bool(d.Get("package_cleanup").(bool))
+		}
+
+		if d.HasChange("reboot_after_user") {
+			input.RebootAfterUse = aws.Bool(d.Get("reboot_after_user").(bool))
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile.go:178]:
```suggestion
		if d.HasChange("reboot_after_use") {
			input.RebootAfterUse = aws.Bool(d.Get("reboot_after_use").(bool))
```

File: internal/service/devicefarm/instance_profile_test.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+		ErrorCheck:   acctest.ErrorCheck(t, devicefarm.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckDeviceFarmInstanceProfileDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDeviceFarmInstanceProfileConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDeviceFarmInstanceProfileExists(resourceName, &profile),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "reboot_after_user", "true"),
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile_test.go:42]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "reboot_after_use", "true"),
```

File: internal/service/devicefarm/instance_profile_test.go [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccDeviceFarmInstanceProfileConfig(rNameUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDeviceFarmInstanceProfileExists(resourceName, &profile),
+					resource.TestCheckResourceAttr(resourceName, "name", rNameUpdated),
+					resource.TestCheckResourceAttr(resourceName, "reboot_after_user", "true"),
```
Comment (ewbankkit) [inline internal/service/devicefarm/instance_profile_test.go:57]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "reboot_after_use", "true"),
```

File: website/docs/r/devicefarm_instance_profile.html.markdown [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+}
+```
+
+## Argument Reference
+
+* `description` - (Optional) The description of the instance profile.
+* `exclude_app_packages_from_cleanup` - (Optional) An array of strings that specifies the list of app packages that should not be cleaned up from the device after a test run.
+* `name` - (Required) The name for the instance profile.
+* `package_cleanup` - (Optional) When set to `true`, Device Farm removes app packages after a test run. The default value is `false` for private devices.
+* `reboot_after_user` - (Optional) When set to `true`, Device Farm reboots the instance after a test run. The default value is `true`.
```
Comment (ewbankkit) [inline website/docs/r/devicefarm_instance_profile.html.markdown:30]:
```suggestion
* `reboot_after_use` - (Optional) When set to `true`, Device Farm reboots the instance after a test run. The default value is `true`.
```

**Review by ewbankkit — CHANGES_REQUESTED** [review]

File: internal/service/devicefarm/instance_profile.go (hunk 1 above)
Comment (DrFaust92) [reply internal/service/devicefarm/instance_profile.go]: 🤦  fixed.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccDeviceFarmInstanceProfile_ PKG=devicefarm
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/devicefarm/... -v -count 1 -parallel 20 -run='TestAccDeviceFarmInstanceProfile_' -timeout 180m
=== RUN   TestAccDeviceFarmInstanceProfile_basic
=== PAUSE TestAccDeviceFarmInstanceProfile_basic
=== RUN   TestAccDeviceFarmInstanceProfile_tags
=== PAUSE TestAccDeviceFarmInstanceProfile_tags
=== RUN   TestAccDeviceFarmInstanceProfile_disappears
=== PAUSE TestAccDeviceFarmInstanceProfile_disappears
=== CONT  TestAccDeviceFarmInstanceProfile_basic
=== CONT  TestAccDeviceFarmInstanceProfile_disappears
=== CONT  TestAccDeviceFarmInstanceProfile_tags
--- PASS: TestAccDeviceFarmInstanceProfile_disappears (13.19s)
--- PASS: TestAccDeviceFarmInstanceProfile_basic (29.70s)
--- PASS: TestAccDeviceFarmInstanceProfile_tags (42.72s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/devicefarm	47.021s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**